otx/cli/utils/tests.py

```diff
@@ -472,3 +474,42 @@ def xfail_templates(templates, xfail_template_ids_reasons):
                 "More than one reason for template. If you have more than one Jira tickets, list them in one reason."
             )
     return xfailed_templates
+
+
+def otx_explain_testing(template, root, otx_dir, args):
+    template_work_dir = get_template_dir(template, root)
+    test_algorithms = ["ActivationMap", "EigenCAM"]
+    check_files = ("Slide1_", "Slide2_", "intel_1_")
+
+    train_ann_file = args.get("--train-ann-file", "")
+    if "hierarchical" in train_ann_file:
+        train_type = "hierarchical"
+    elif "multilabel" in train_ann_file:
+        train_type = "multilabel"
+    else:
+        train_type = "default"
+
+    for test_algorithm in test_algorithms:
+        save_dir = f"explain_{template.model_template_id}/{test_algorithm}/{train_type}/"
+        output_dir = os.path.join(template_work_dir, save_dir)
+        compare_dir = os.path.join(f"{otx_dir}/data/explain_samples/", save_dir)
+        command_line = [
+            "otx",
+            "explain",
+            template.model_template_path,
+            "--load-weights",
+            f"{template_work_dir}/trained_{template.model_template_id}/weights.pth",
+            "--explain-data-root",
+            os.path.join(otx_dir, args["--input"]),
+            "--save-explanation-to",
+            output_dir,
+            "--explain-algorithm",
+            test_algorithm,
+        ]
+        assert run(command_line).returncode == 0
+        for fname in os.listdir(output_dir):
+            if fname.startswith(check_files) and "overlay" in fname:
```

Comment:
As discussed, to verify the explanation algorithms, let's compare raw saliency maps. Typically it is just 7*7 uint8 ndarray. This will help to avoid storing the full res images.

As for the saliancy map postprocessing (resize, colormap, etc.), let's have a separate unit test for it with some dummy single low-res image.

Reply:
As we've discussed, we can reduce saving memory by only saving the 7x7 original map as a gray image or binary, but actually, the sample image storage size is not that big, and we should change the `saliency_map` saving algorithm to save the raw image, which can be harmful to existing features. I mean, Do we really need this?
We can remove `Slide2_` images so that we can change the total images to 2MB -> 1.2MB

Reply:
I would propose to keep the `get_actmap` as-is. Please use for example [this](https://github.com/openvinotoolkit/training_extensions/blob/aba492c18bdbbf83b90316de0325bdaeeeff634f/external/model-preparation-algorithm/tests/test_xai.py) test implementation as a reference.

Reply:
It's using MPA source, but OTX test codes are using a CLI to test whether the CLI command works fine. It's detouring using the OTX CLI, which can't test the whole CLI process. Why don't we just keep it as it was? We're too time-consuming in modifying a test code, which works fine. We can go ahead to support det models or OV and modify the test code later.

Reply:
It's using MPA source, but OTX test codes are using a CLI to test whether the CLI command works fine. It's detouring using the OTX CLI, which can't test the whole CLI process. Why don't we just keep it as it was?

Reply:
I agree. Let's keep it as it was for this PR